showcase/src/app/welcome/pages/reactive-form-errors/reactive-form-errors-page.component.html

```diff
@@ -0,0 +1,231 @@
+<div class="container">
+	<h1 class="mat-display-1" translate>
+		SHOWCASE.NGX_FORM_ERRORS.TITLE
+	</h1>
+	<h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
+	<section class="stark-section">
+		<example-viewer
+			id="example"
+			[extensions]="['HTML', 'SCSS', 'TS']"
+			filesPath="reactive-form-errors/reactive-form-errors"
+			[exampleTitle]="'SHOWCASE.NGX_FORM_ERRORS.EXAMPLE' | translate"
+		>
+			<form [formGroup]="formGroup" (ngSubmit)="onSubmitUserDetails(formGroup)">
+				<div fxLayout="column" fxLayoutAlign="space-between" fxLayoutGap="10px">
+					<app-card fxFlex="1 1 auto" class="form-card">
+						<mat-card-content>
+							<mat-form-field class="full-width" [ngClass]="getErrorClass('username')">
+								<input
+									matInput
+									[placeholder]="'SHOWCASE.NGX_FORM_ERRORS.FIELDS.USER_NAME' | translate"
+									formControlName="username"
+									required
+									autocomplete="off"
+								/>
+								<mat-error>
+									<ng-template ngxFormErrors="username" #usernameField="ngxFormErrors"></ng-template>
+								</mat-error>
+							</mat-form-field>
+
+							<div formGroupName="matchingPasswords" ngxFormErrorsGroup="passwords-group">
+								<mat-form-field class="full-width" [ngClass]="getErrorClass('matchingPasswords.password')">
+									<input
+										matInput
+										type="password"
+										[placeholder]="'SHOWCASE.NGX_FORM_ERRORS.FIELDS.PASSWORD' | translate"
+										formControlName="password"
+										required
+										autocomplete="off"
+									/>
+									<mat-error>
+										<ng-template
+											ngxFormErrors="matchingPasswords.password"
+											ngxFormErrorsFieldName="SHOWCASE.NGX_FORM_ERRORS.FIELDS.ALIAS.PASSWORD_ALIAS"
+											#passwordField="ngxFormErrors"
+										></ng-template>
+									</mat-error>
+								</mat-form-field>
+
+								<mat-form-field class="full-width" [ngClass]="getErrorClass('matchingPasswords.confirmPassword')">
+									<input
+										matInput
+										type="password"
+										[placeholder]="'SHOWCASE.NGX_FORM_ERRORS.FIELDS.CONFIRM_PASSWORD' | translate"
+										formControlName="confirmPassword"
+										[errorStateMatcher]="parentErrorStateMatcher"
```

Comment:
I did some tests locally and I have the feeling this line should not be there. Indeed, if I remove it, I don't have the following issue:
![image](https://user-images.githubusercontent.com/10333565/77437818-022e2500-6ddd-11ea-85b2-5dea5677fcd6.png)

(password fields contain the same values and respect the rules)

Could you please explain why this "parentErrorStateMatcher" is necessary ? 😊 

Another point, if the passwords are not equal, the "Submit" button is disabled but no error is displayed 😕

Reply:
Yeah, I was a bit lazy in the original implementation and I didn't check properly that it was working as expected.

The reason of the `StateMatcher` is to programmatically tell the `mat-input` when to toggle to the "error" state because we use a form group validator instead of a control validator, so we must tell the field when there is an error in the group. However our state matcher implementation is indeed buggy.

So I reworked this and the `StateMatcher` is not needed anymore 👍  I now use simple control validators and I even use the literal `SHOWCASE.NGX_FORM_ERRORS.FORM.VALIDATION.CONFIRM_PASSWORD.ARE_EQUAL` which was not used at all before 😉 

Now it works like a charm 💪